js/editor.js

```diff
@@ -54,6 +54,35 @@
         }
     }
 
+    /**
+     * Interrupts the default click event on external links inside
+     * the shadow dom and opens them in a new tab instead
+     * @param {Array} externalLinks - all external links inside the shadow dom
+     */
+    function openLinksInNewTab(externalLinks) {
+        externalLinks.forEach(function(externalLink) {
+            externalLink.addEventListener('click', function(event) {
+                event.preventDefault();
+                window.open(externalLink.href);
+            });
+        });
+    }
+
+    /**
+     * Interrupts the default click event on relative links inside
+     * the shadow dom and scrolls to the targeted anchor
+     * @param {Object} shadow - the shadow dom root
+     * @param {Array} relativeLinks - all relative links inside the shadow dom
+     */
+    function scrollToAnchors(shadow, relativeLinks) {
+        relativeLinks.forEach(function(relativeLink) {
+            relativeLink.addEventListener('click', function(event) {
+                event.preventDefault();
+                shadow.querySelector(relativeLink.hash).scrollIntoView();
+            });
+        });
+    }
```

Comment:
I would suggest adding these two functions to our utilities module:
https://github.com/stephanmax/interactive-examples/blob/e669e9f9f637684d371f97ac77ff7dbe27bf7145/js/editor-libs/mce-utils.js